src/tools/rbd_mirror/ImageReplayer.cc

```diff
@@ -501,15 +501,19 @@ void ImageReplayer<I>::bootstrap() {
   Context *ctx = create_context_callback<
     ImageReplayer, &ImageReplayer<I>::handle_bootstrap>(this);
 
-  BootstrapRequest<I> *request = BootstrapRequest<I>::create(
-    m_threads, *m_local_ioctx, m_remote_image.io_ctx, m_instance_watcher,
-    &m_local_image_ctx, m_local_image_id, m_remote_image.image_id,
-    m_global_image_id, m_local_mirror_uuid, m_remote_image.mirror_uuid,
-    m_remote_journaler, &m_client_state, &m_client_meta, ctx,
-    &m_resync_requested, &m_progress_cxt);
-
+  BootstrapRequest<I> *request = nullptr;
   {
     Mutex::Locker locker(m_lock);
+    if (on_start_interrupted(m_lock)) {
+      return;
```

Comment:
Need to move `ctx` creation below this return. I suppose this is the cause of the observed valgrind error:

 http://qa-proxy.ceph.com/teuthology/trociny-2019-02-25_13:52:38-rbd-wip-mgolub-testing-distro-basic-smithi/3636974/teuthology.log

Reply:
Indeed -- thx. Fix pushed.